app/store/migrations/056.ts

```diff
@@ -0,0 +1,31 @@
+import { captureException } from '@sentry/react-native';
+import { isObject } from '@metamask/utils';
+import { ensureValidState } from './util';
+
+export default function migrate(state: unknown) {
+  if (!ensureValidState(state, 56)) {
+    // Increment the migration number as appropriate
+    return state;
+  }
+
+  const preferencesController =
+    state.engine.backgroundState.PreferencesController;
+
+  if (!isObject(preferencesController)) {
+    captureException(
+      new Error(
+        `FATAL ERROR: Migration 56: Invalid PreferencesController state error: '${typeof preferencesController}'`,
+      ),
+    );
+    return state;
+  }
+
+  const decommisionedIpfsGateway = 'https://cloudflare-ipfs.com/ipfs/';
+  const newDefaultIpfsGateway = 'https://dweb.link/ipfs/';
+
+  if (decommisionedIpfsGateway === preferencesController.ipfsGateway) {
```

Comment:
I think it's good if we add a conditional operator
```suggestion
  if (decommisionedIpfsGateway === preferencesController?.ipfsGateway) {
```